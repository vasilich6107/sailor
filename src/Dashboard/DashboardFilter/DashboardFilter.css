
.filter {
    display: flex;
    justify-content: space-evenly;
    padding: 20px;
}

.filter__item {
    display: flex;
}