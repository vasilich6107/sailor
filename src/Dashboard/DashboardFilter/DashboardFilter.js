// @flow
import React from 'react';
import './DashboardFilter.css';
import type { DashboardFilterType } from '../../types/Dashboard';

type Props = {
  initialValues: DashboardFilterType,
  onFilterChange: (name: string) => (e: SyntheticInputEvent<HTMLInputElement>) => {},
};

const DashboardFilter = ({ initialValues, onFilterChange }: Props) => (
  <div className="filter">
    <div className="filter__item">
      <div>Name</div>
      <div>
        <input
          id="nameInput"
          type="text"
          placeholder="First or Last name"
          value={initialValues.name}
          onChange={onFilterChange('name')}
        />
      </div>
    </div>
    <div className="filter__item">
      <div>Country</div>
      <div>
        <input
          id="cityInput"
          type="text"
          placeholder="Country"
          value={initialValues.city}
          onChange={onFilterChange('city')}
        />
      </div>
    </div>
  </div>
);

export default DashboardFilter;
