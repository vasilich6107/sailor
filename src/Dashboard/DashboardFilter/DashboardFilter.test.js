import React from 'react';
import { shallow } from 'enzyme';
import DashboardFilter from './DashboardFilter';

it('Renders filter with empty initial values', () => {
  const handleFilterChange = jest.fn().mockImplementation(() => jest.fn());

  const emptyInitialValues = {
    name: '',
    city: '',
  };
  expect(
    shallow(<DashboardFilter initialValues={emptyInitialValues} onFilterChange={handleFilterChange} />)
  ).toMatchSnapshot();

  expect(handleFilterChange.mock.calls.length).toBe(2);
  expect(handleFilterChange.mock.calls[0][0]).toBe('name');
  expect(handleFilterChange.mock.calls[1][0]).toBe('city');
});

it('Renders filter with prefilled initial values', () => {
  const handleFilterChange = jest.fn().mockImplementation(() => jest.fn());

  const prefilledInitialValues = {
    name: 'some',
    city: 'city',
  };
  expect(
    shallow(<DashboardFilter initialValues={prefilledInitialValues} onFilterChange={handleFilterChange} />)
  ).toMatchSnapshot();

  expect(handleFilterChange.mock.calls.length).toBe(2);
  expect(handleFilterChange.mock.calls[0][0]).toBe('name');
  expect(handleFilterChange.mock.calls[1][0]).toBe('city');
});

it('Renders filter and calls the onChange function', () => {
  const inputOnChange = jest.fn();
  const handleFilterChange = jest.fn().mockImplementation(() => inputOnChange);

  const emptyInitialValues = {
    name: '',
    city: '',
  };

  const component = shallow(<DashboardFilter initialValues={emptyInitialValues} onFilterChange={handleFilterChange} />);
  expect(component).toMatchSnapshot();

  const nameValue = { target: { value: 'fr' } };
  const cityValue = { target: { value: 'ba' } };

  component.find('#nameInput').simulate('change', nameValue);
  expect(component).toMatchSnapshot();

  expect(inputOnChange.mock.calls.length).toBe(1);
  expect(inputOnChange.mock.calls[0][0]).toBe(nameValue);

  component.find('#cityInput').simulate('change', cityValue);
  expect(component).toMatchSnapshot();

  expect(inputOnChange.mock.calls.length).toBe(2);
  expect(inputOnChange.mock.calls[1][0]).toBe(cityValue);
});
