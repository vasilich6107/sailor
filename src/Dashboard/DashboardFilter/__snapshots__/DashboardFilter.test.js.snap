// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders filter and calls the onChange function 1`] = `
<div
  className="filter"
>
  <div
    className="filter__item"
  >
    <div>
      Name
    </div>
    <div>
      <input
        id="nameInput"
        onChange={[Function]}
        placeholder="First or Last name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="filter__item"
  >
    <div>
      Country
    </div>
    <div>
      <input
        id="cityInput"
        onChange={[Function]}
        placeholder="Country"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Renders filter and calls the onChange function 2`] = `
<div
  className="filter"
>
  <div
    className="filter__item"
  >
    <div>
      Name
    </div>
    <div>
      <input
        id="nameInput"
        onChange={[Function]}
        placeholder="First or Last name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="filter__item"
  >
    <div>
      Country
    </div>
    <div>
      <input
        id="cityInput"
        onChange={[Function]}
        placeholder="Country"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Renders filter and calls the onChange function 3`] = `
<div
  className="filter"
>
  <div
    className="filter__item"
  >
    <div>
      Name
    </div>
    <div>
      <input
        id="nameInput"
        onChange={[Function]}
        placeholder="First or Last name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="filter__item"
  >
    <div>
      Country
    </div>
    <div>
      <input
        id="cityInput"
        onChange={[Function]}
        placeholder="Country"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Renders filter with empty initial values 1`] = `
<div
  className="filter"
>
  <div
    className="filter__item"
  >
    <div>
      Name
    </div>
    <div>
      <input
        id="nameInput"
        onChange={[Function]}
        placeholder="First or Last name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="filter__item"
  >
    <div>
      Country
    </div>
    <div>
      <input
        id="cityInput"
        onChange={[Function]}
        placeholder="Country"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Renders filter with prefilled initial values 1`] = `
<div
  className="filter"
>
  <div
    className="filter__item"
  >
    <div>
      Name
    </div>
    <div>
      <input
        id="nameInput"
        onChange={[Function]}
        placeholder="First or Last name"
        type="text"
        value="some"
      />
    </div>
  </div>
  <div
    className="filter__item"
  >
    <div>
      Country
    </div>
    <div>
      <input
        id="cityInput"
        onChange={[Function]}
        placeholder="Country"
        type="text"
        value="city"
      />
    </div>
  </div>
</div>
`;
